lddwrap/main.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+"""Command-line tool entry point for lddwrap to determine shared libraries required by a program."""
+
+# This file is necessary so that we can specify the entry point for pex.
+
+import argparse
+import pathlib
+import sys
+from typing import List, Any, TextIO
+
+import lddwrap
+import pylddwrap_meta
+
+
+class Args:
+    """Represent parsed command-line arguments."""
+
+    def __init__(self, args: Any) -> None:
+        """Initialize with arguments parsed with ``argparse``."""
+        self.version = bool(args.version)
+        self.format = str(args.format)
+        self.path = pathlib.Path(args.path)
+
+
+def parse_args(sys_argv: List[str]) -> Args:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "--version",
+        help="Display the version and return immediately",
+        action='store_true')
+    parser.add_argument(
+        "--format",
+        help="Specify the output format. [NEED PATH]",
+        default='verbose',
+        choices=['verbose', 'json'])
+
+    parser.add_argument(
+        "path", help="Specify path to the binary", nargs='?', default='')
+
+    args = parser.parse_args(sys_argv[1:])
+
+    if not args.version and not pathlib.Path(args.path).is_file():
+        parser.error(
+            "Either path '{}' is not given or is invalid. Check out --help for more information."
```

Comment:
Please separate the checks and output two different error messages. It is confusing for the user to deal with the logical OR.

You actually have three cases: 1) file not given, 2) file does not exist, 3) file not a file (but a directory).